header.go

```diff
@@ -124,3 +102,32 @@ func setHeaderCanonical(h http.Header, key, value string) {
 func delHeaderCanonical(h http.Header, key string) {
 	delete(h, key)
 }
+
+func isProtocolHeader(key string) bool {
```

Comment:
FYI, this list is long enough that it may be worth benchmarking this approach vs a package map var. So instead of needing ~20 string comparisons before deciding a key is _not_ a protocol header, it could be done with a single map query.

Reply:
Made it a map.